pkg/splunk/enterprise/util.go

```diff
@@ -287,11 +298,12 @@ func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.Me
 		}
 	}
 
-	if localAppsConf != yamlConfHeader && localAppsConf != yamlConfLocalHeader {
+	// Don't update the configMap if there is nothing to write.
+	if localAppsConf != yamlConfLocalHeader {
 		mapAppListing["app-list-local.yaml"] = localAppsConf
 	}
 
-	if clusterAppsConf != yamlConfHeader {
+	if clusterAppsConf != yamlConfIdcHeader || clusterAppsConf != yamlConfShcHeader || clusterAppsConf != "" {
```

Comment:
Doesn't it supposed to be logical &&, for all the 3 conditions?

Reply:
Yeah good catch, I think you are right :)